File: 1128-remove-all-adjacent-duplicates-in-string/1128-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        stack<char> st;
        string result = "";
        for(int i=0;i<s.size();i++)
        {
            if(st.empty() ||  st.top()!=s[i])
            {
                st.push(s[i]);
            }
            else{
                st.pop();
            }
        }

        while(!st.empty())
        {
            result +=st.top();
            st.pop();
        }
         reverse(result.begin(),result.end());
        return result;

    }
};

File: 1128-remove-all-adjacent-duplicates-in-string/1128-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> st = new Stack<>();

        StringBuilder result = new StringBuilder();

        for(int i=0;i<s.length();i++)
        {
            if(st.empty() || st.peek()!= s.charAt(i))
            {
                st.push(s.charAt(i));
            }
            else{
                st.pop();
            }
        }

        while(!st.empty())
        {
            result.append(st.pop());
        }

        return result.reverse().toString();

      
    }
}